### PR TITLE
allow vault_pki renewal interval to be configured by VaultLeaseRenewalThreshold

### PR DESCRIPTION
**Pull request by kevinschoonover**
closes https://github.com/hashicorp/consul-template/issues/1646

### CODE REVIEW

**Comment by VioletHynes** [comment]
Hey! Thanks for this contribution. I'm not currently on duty to review community PRs, but I've alerted the person who is and we should get eyes on this soon once they wrap up what they're currently looking at. 

I've really appreciated the high quality PRs you've been putting up recently, and how quick you've been to respond to feedback, and I'm super excited to get all of this merged. If one of your PRs ever goes too long without attention or love, feel free to tag me directly and I'll try and make sure someone gets eyes on it.

Once we get this merged, we'll work on getting a new consul-template version and getting it updated in Vault Agent, too.

File: dependency/vault_pki.go [hunk 1]
```diff
@@ -127,9 +127,12 @@ func goodFor(cert *x509.Certificate) (time.Duration, bool) {
...
 	lifespanMilliseconds := lifespanDur.Milliseconds()
-	// calculate the 'time the certificate should be rotated' by figuring out
-	// 87-93% of the lifespan and adding it to the start
-	rotationTime := start.Add(time.Millisecond * time.Duration(((lifespanMilliseconds*9)/10)+(lifespanMilliseconds*int64(r.Intn(6)-3))/100))
+	// calculate the 'time the certificate should be rotated' by figuring out -2%
+	// - 3% + VaultLeaseRenewalThreshold of the lifespan and adding it to the
+	// start
+	rotationTime := start.Add(time.Millisecond * time.Duration(
+		float64(lifespanMilliseconds)*VaultLeaseRenewalThreshold+float64(lifespanMilliseconds*(int64(r.Intn(6)-3)/100.0)),
+	))
```
Comment (divyaac) [inline dependency/vault_pki.go:135]: There's a linter failure having to do with conversions that might be worth checking.

Comment (divyaac) [reply dependency/vault_pki.go:135]: There is also another linter failure on line [dependency/vault_pki.go#L136](https://github.com/hashicorp/consul-template/pull/1872/files#annotation_20734928620) ] that we forgot to correct in the last PR. If we could fix that, that would be amazing!

File: dependency/vault_pki.go [hunk 2]
```diff
@@ -127,9 +127,12 @@ func goodFor(cert *x509.Certificate) (time.Duration, bool) {
 	lifespanDur := end.Sub(start)
 	r := rand.New(rand.NewSource(time.Now().UnixNano()))
 	lifespanMilliseconds := lifespanDur.Milliseconds()
-	// calculate the 'time the certificate should be rotated' by figuring out
-	// 87-93% of the lifespan and adding it to the start
-	rotationTime := start.Add(time.Millisecond * time.Duration(((lifespanMilliseconds*9)/10)+(lifespanMilliseconds*int64(r.Intn(6)-3))/100))
+	// calculate the 'time the certificate should be rotated' by figuring out -2%
+	// - 3% + VaultLeaseRenewalThreshold of the lifespan and adding it to the
```
Comment (divyaac) [inline dependency/vault_pki.go:131]: Apologies of I'm being obtuse - it seems that we are still calculating 87-93% of the VaultLeaseRenewalThresdhold (not -2% - 3% right)?

**Review by divyaac — CHANGES_REQUESTED** [review]

File: dependency/vault_pki.go (hunk 1 above)
Comment (divyaac) [inline dependency/vault_pki.go:135]: There's a linter failure having to do with conversions that might be worth checking. There is also another linter failure on line [dependency/vault_pki.go#L136](https://github.com/hashicorp/consul-template/pull/1872/files#annotation_20734928620) ] that we forgot to correct in the last PR. If we could fix that, that would be amazing!

**Review by divyaac — CHANGES_REQUESTED** [review]

**Comment by divyaac** [comment]
There is also a failing test that might be worth checking if it's caused by these changes!

File: dependency/vault_pki.go (hunk 2 above)
Comment (kevinschoonover) [reply dependency/vault_pki.go:131]: My understanding of the math (using .80 as an example to make it easier for me)
+ float64(lifespanMilliseconds)*VaultLeaseRenewalThreshold = float64(lifespanMilliseconds) * .80
+ float64(lifespanMilliseconds*(int64(r.Intn(6)-3)/100.0)) = float64(lifespanMilliseconds) * [-.03, .03]

`float64(lifespanMilliseconds) * .80 +  float64(lifespanMilliseconds) * [-.03, .03]` - which would come out to .77 - .83 of lifespanMilliseconds which is -3, +3 of VaultRenewalThreshold

Comment (divyaac) [reply dependency/vault_pki.go:131]: Ah yes sorry makes sense... It might make sense to update that comment to show "-3, +3" instead of "-3, -3"

Comment (kevinschoonover) [reply dependency/vault_pki.go]: I double checked and r.Intn() is actually [0, n) so it would be [-3, 2]. I updated the code

**Comment by kevinschoonover** [comment]
> There is also a failing test that might be worth checking if it's caused by these changes!

It looks like these are unrelated and I confirmed they pass locally. The first time the test only failed in enterprise because the race? test took too long. In the latest version its mad about TestFileQuery_Fetch/fires_changes which im not sure i touched

File: dependency/vault_pki.go [hunk 3]
```diff
@@ -127,13 +127,16 @@ func goodFor(cert *x509.Certificate) (time.Duration, bool) {
 	lifespanDur := end.Sub(start)
 	r := rand.New(rand.NewSource(time.Now().UnixNano()))
 	lifespanMilliseconds := lifespanDur.Milliseconds()
-	// calculate the 'time the certificate should be rotated' by figuring out
-	// 87-93% of the lifespan and adding it to the start
-	rotationTime := start.Add(time.Millisecond * time.Duration(((lifespanMilliseconds*9)/10)+(lifespanMilliseconds*int64(r.Intn(6)-3))/100))
+	// calculate the 'time the certificate should be rotated' by figuring out -3%
+	// - 2% + VaultLeaseRenewalThreshold of the lifespan and adding it to the
```
Comment (divyaac) [inline dependency/vault_pki.go:131]: We should probably update this comment to "-3, +3"

**Review by divyaac — CHANGES_REQUESTED** [review]

**Review by divyaac — APPROVED** [review]

**Pull request merged**